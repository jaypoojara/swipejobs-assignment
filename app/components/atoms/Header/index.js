import React from 'react';
import { StyleSheet, Text, View, TouchableOpacity, Image } from 'react-native';
import { Appbar } from 'react-native-paper';
import { FontAwesome5 } from '@expo/vector-icons';
import NavigationService from '../../../services/NavigationService';
import colors from '../../../themes/colors';
import images from '../../../themes/images';
import T from '../T';

const Header = ({ name, fromProfile = false }) => {
  return (
    <Appbar.Header style={{ backgroundColor: colors.black }}>
      {fromProfile ? (
        <View style={styles.profileHeaderContainer}>
          <TouchableOpacity onPress={() => NavigationService.goBack()}>
            <FontAwesome5 name="chevron-left" size={24} color="white" />
          </TouchableOpacity>
          <T style={[styles.profileTitle]} text={'Jim Rose'} />
        </View>
      ) : (
        <View style={styles.headerContainer}>
          <Image
            source={images.swipeLogo}
            resizeMode="contain"
            style={styles.headerLogo}
          />
          <TouchableOpacity
            onPress={() => NavigationService.navigate('ProfileScreen')}
          >
            {name && <T style={[styles.titleText]} text={name} />}
          </TouchableOpacity>
        </View>
      )}
    </Appbar.Header>
  );
};

export default Header;

const styles = StyleSheet.create({
  headerContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginHorizontal: 10
  },
  headerLogo: {
    height: 50,
    width: '50%',
    tintColor: colors.white
  },
  titleText: {
    color: colors.white,
    fontSize: 20
  },
  profileHeaderContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    marginHorizontal: 10
  },
  profileTitle: {
    color: 'white',
    flex: 1,
    textAlign: 'center',
    marginRight: 10,
    fontSize: 20
  }
});
