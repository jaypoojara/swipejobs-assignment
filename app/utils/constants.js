export const NOTIFICATION_TYPE = {
  DEFAULT: 0,
  SUCCESS: 1,
  ERROR: 2,
  WARNING: 3,
  INFO: 4
};

export const WORKER_ID = '7f90df6e-b832-44e2-b624-3143d428001f';
