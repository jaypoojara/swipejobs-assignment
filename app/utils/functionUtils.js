import moment from 'moment';
import momentTz from 'moment-timezone';

export const getDuration = (startDate, endDate) => {
  const days = moment(startDate)
    .local()
    .format('MMM DD, ddd HH:mm A');
  const endDateTemp = moment(endDate)
    .local()
    .format('HH:mm A');
  const timeZone = momentTz.tz.guess();
  const timeZoneAbr = momentTz.tz(timeZone).zoneAbbr();
  return `${days} - ${endDateTemp} ${timeZoneAbr}`;
};
