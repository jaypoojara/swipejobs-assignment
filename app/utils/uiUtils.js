import { showMessage } from 'react-native-flash-message';
import { NOTIFICATION_TYPE } from './constants';
export function showMessageToUser({ title, message, notificationType }) {
  let messageType;
  switch (notificationType) {
    case NOTIFICATION_TYPE.DEFAULT:
      messageType = 'default';
      break;
    case NOTIFICATION_TYPE.INFO:
      messageType = 'info';
      break;
    case NOTIFICATION_TYPE.SUCCESS:
      messageType = 'success';
      break;
    case NOTIFICATION_TYPE.ERROR:
      messageType = 'danger';
      break;
    case NOTIFICATION_TYPE.WARNING:
      messageType = 'warning';
      break;
    default:
      messageType = 'none';
  }
  setTimeout(() => {
    showMessage({
      message: title,
      description: message,
      type: messageType
    });
  }, 100);
}
