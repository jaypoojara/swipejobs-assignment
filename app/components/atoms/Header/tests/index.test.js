import React from 'react';
import { renderProvider } from '../../../../utils/testUtils';
import Header from '../index';

describe('<Header /> Component tests', () => {
  it('should render and match the header snapshot', () => {
    const baseElement = renderProvider(<Header fromProfile={false} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render and match the header snapshot when profile is true', () => {
    const baseElement = renderProvider(<Header fromProfile={true} />);
    expect(baseElement).toMatchSnapshot();
  });
});
