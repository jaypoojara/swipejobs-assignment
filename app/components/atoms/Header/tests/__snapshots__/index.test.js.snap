// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Component tests should render and match the header snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "elevation": 4,
        "zIndex": 0,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": 0,
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 10,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../app/assets/images/swipejobs_logo.png",
          }
        }
        style={
          Object {
            "height": 50,
            "tintColor": "#ffffff",
            "width": "50%",
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`<Header /> Component tests should render and match the header snapshot when profile is true 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "elevation": 4,
        "zIndex": 0,
      },
      Object {
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 4,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "elevation": 0,
        "flexDirection": "row",
        "height": 56,
        "marginTop": 0,
        "paddingHorizontal": 4,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "marginHorizontal": 10,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "flex": 1,
              "fontSize": 20,
              "marginRight": 10,
              "textAlign": "center",
            },
          ]
        }
        testID="t"
      >
        Jim Rose
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
